packages/perseus-editor/src/components/locked-vector-settings.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * LockedVectorSettings is a component that allows the user to edit the
+ * settings of specifically a locked vector on the graph.
+ *
+ * Used in the interactive graph editor's locked figures section.
+ */
+import {vector as kvector} from "@khanacademy/kmath";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import {spacing, color as wbColor} from "@khanacademy/wonder-blocks-tokens";
+import {LabelLarge, LabelMedium} from "@khanacademy/wonder-blocks-typography";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+import {useState} from "react";
+
+import ColorSelect from "./color-select";
+import CoordinatePairInput from "./coordinate-pair-input";
+import LineSwatch from "./line-swatch";
+import LockedFigureSettingsAccordion from "./locked-figure-settings-accordion";
+import LockedFigureSettingsActions from "./locked-figure-settings-actions";
+
+import type {AccordionProps} from "./locked-figure-settings";
+import type {
+    Coord,
+    LockedFigure,
+    LockedFigureColor,
+    LockedVectorType,
+} from "@khanacademy/perseus";
+
+const lengthErrorMessage = "The vector cannot have length 0.";
+
+export type Props = LockedVectorType &
+    AccordionProps & {
+        /**
+         * Called when the delete button is pressed.
+         */
+        onRemove: () => void;
+        /**
+         * Called when the props (points, color, etc.) are updated.
+         */
+        onChangeProps: (newProps: Partial<LockedFigure>) => void;
+    };
+
+const LockedVectorSettings = (props: Props) => {
+    const [tailCoordExpanded, setTailCoordExpanded] = useState(false);
+    const [tipCoordExpanded, setTipCoordExpanded] = useState(false);
+    const {points, color: lineColor, onChangeProps, onRemove} = props;
+    const [tail, tip] = points;
+    const lineLabel = `Vector (${tail[0]}, ${tail[1]}), (${tip[0]}, ${tip[1]})`;
+
+    // Check if the line has length 0.
+    const isInvalid = kvector.equal(tail, tip);
+
+    function handleChangePoint(newCoord: Coord | undefined, index: 0 | 1) {
+        if (typeof newCoord !== "undefined") {
+            const newPoints = [...points] as [Coord, Coord];
```

Comment:
Can you change the `as` to a `satisfies`? Apparently `as` just blows away the type altogether and then TypeScript doesn't do proper typechecking.

Also, I think you can use `CollinearTuple` instead of `[Coord, Coord]`?

Reply:
I'm wondering about the level of clarity that using `CollinearTuple` has compared to the current `points: [tail: Coord, tip: Coord]` ([current type definition](https://github.com/Khan/perseus/blob/1a070674ea55fe61b2d3e878f6d6f94af8ebe981/packages/perseus/src/perseus-types.ts#L708)). It's not just two coordinates - each has a specific meaning.